frontend/src/Admin/components/ShowJudgesTab.js

```diff
@@ -1,5 +1,90 @@
 import React from 'react'
+import PropTypes from 'prop-types'
+import { Row, Col, Progress, Button } from 'reactstrap'
+import { Link } from 'react-router-dom'
 
-export default function () {
-  return 'Judges'
+// Calculates a Bootstrap color for the given fractional progress, given
+// as a number from 0 to 1.
+const colorForProgress = progress => {
+  if (progress < 0.1) {
+    return 'danger'
+  }
+  if (progress < 0.25) {
+    return 'warning'
+  }
+  if (progress < 1) {
+    return null // default color
+  }
+  return 'success'
 }
+
+const ShowJudgesTab = ({ show }) => {
+  // Extract all votes we care about
+  const allVotes = show.entries.reduce(
```

Comment:
This section until `const numEntries` should be moved up into a container. The component should only worry about how the progress bars are rendered, not how the data should be formatted first. Pass down that formatted data as `props`.